modules/widgets/migrate-to-core/image-widget.php

```diff
@@ -56,7 +107,29 @@ function jetpack_migrate_image_widget() {
 			continue;
 		}
 
-		$media_image[ $id ] = array_merge( $default_data, array_intersect_key( $widget, $default_data ), array(
+		// Ensure widget has no keys other than those expected.
+		// Not all widgets have conditions, so lets add it in.
+		$widget_copy = array_merge( array( 'conditions' => null ), $widget );
+		$non_whitelisted_keys = array_diff_key( $widget_copy, array(
+			'title' => '',
+			'img_url' => '',
+			'alt_text' => '',
+			'img_title' => '',
+			'caption' => '',
+			'align' => '',
+			'img_width' => '',
+			'img_height' => '',
+			'link' => '',
+			'link_target_blank' => '',
+			'conditions' => '',
+		) );
+
+		if ( count( $non_whitelisted_keys ) > 0 ) {
+			// skipping the widget in question
+			continue;
+		}
+
+		$media_image[ $id ] = array_merge( $default_data, $widget, array(
```

Comment:
If `$media_image[ $id ]` is undefined, it will get assigned, if it is defined, it will get updated, or am I reading something wrong?